modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java

```diff
@@ -659,13 +659,15 @@ protected final String startInfo() {
     /**
      * Stops this cache. Child classes should override this method
      * to provide custom stop behavior.
+     *
+     * @param destroy Destroy data flag. Setting to <code>true</code> will remove all cache data.
      */
-    public void stop() {
+    public void stop(boolean destroy) {
         // Nulling thread local reference to ensure values will be eventually GCed
         // no matter what references these futures are holding.
         lastFut = null;
 
-        if (!ctx.kernalContext().isStopping())
+        if (!ctx.kernalContext().isStopping() && destroy)
```

Comment:
Could be just removed (condition and statement under it). See my comment to `GridCacheProcessor.stopCache()` method.